unit_testing/test_labels.py

```diff
@@ -0,0 +1,208 @@
+    shape = (1,2,3)
+    """
+    i = fake_3dimage2()
+    img = Image(i.get_data(), hdr=i.header,
+                orientation="RPI",
+                dim=i.header.get_data_shape(),
+                )
+    return img
+
+
+def fake_3dimage_sct():
+    """
+    :return: an Image (3D) in RAS+ (aka SCT LPI) space
+    shape = (7,8,9)
+    """
+    i = fake_3dimage()
+    img = Image(i.get_data(), hdr=i.header,
+                orientation="LPI",
+                dim=i.header.get_data_shape(),
+                )
+    return img
+
+
+test_images = [fake_3dimage_sct(), fake_3dimage_sct2(), t2_img]
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_create_labels_empty(test_image):
+    a = test_image.copy()
+    expected = zeros_like(a)
+
+    labels = [Coordinate(l) for l in [[0, 0, 0, 7], [0, 1, 2, 5]]]
+    expected.data[0, 0, 0] = 7
+    expected.data[0, 1, 2] = 5
+
+    b = sct_labels.create_labels_empty(a, labels)
+
+    diff = b.data == expected.data
+    assert diff.all()
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_create_labels(test_image):
+    a = test_image.copy()
+    labels = [Coordinate(l) for l in [[0, 1, 0, 99], [0, 1, 2, 5]]]
+
+    b = sct_labels.create_labels(a, labels)
+
+    assert b.data[0, 1, 0] == 99
+    assert b.data[0, 1, 2] == 5
+
+
+@pytest.mark.parametrize("test_seg", [seg_img])
+def test_create_labels_along_segmentation(test_seg):
+    a = test_seg.copy()
+    labels = [(5, 1), (14, 2), (23, 3)]
+
+    b = sct_labels.create_labels_along_segmentation(a, labels)
+
+    assert b.orientation == a.orientation
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_cubic_to_point(test_image):
+    a = test_image.copy()
+    sct_labels.cubic_to_point(a)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_increment_z_inverse(test_image):
+    a = test_image.copy()
+    sct_labels.increment_z_inverse(a)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_seg,test_labels", [(seg_img, labels_img)])
+def test_labelize_from_disks(test_seg, test_labels):
+    seg = test_seg.copy()
+    ref = test_labels.copy()
+
+    sct_labels.labelize_from_disks(seg, ref)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_label_vertebrae(test_image):
+    a = test_image.copy()
+    sct_labels.label_vertebrae(a)
+    sct_labels.label_vertebrae(a, [1, 2, 3, 4])
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_compute_mean_squared_error(test_image):
+    src = test_image.copy()
+    ref = test_image.copy()
+
+    for z in range(3):
+        for y in range(2):
+            for x in range(1):
+                ref.data[x, y, z] *= 10
+
+    sct_labels.compute_mean_squared_error(src, ref)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.skip(reason="Too long to run on large image!")
```

Comment:
indeed, i've also noticed that. The algorithm is sub-optimal and should be changed (not in this PR of course)